lib/node_modules/@stdlib/lapack/base/dlascl/lib/base.js

```diff
@@ -0,0 +1,285 @@
+* @param {NonNegativeInteger} KL - lower band width of `A`. Referenced only if type is `symmetric-banded-lower` or `banded`.
+* @param {NonNegativeInteger} KU - upper band width of `A`. Referenced only if type is `symmetric-banded-upper` or `banded`.
+* @param {number} CFROM - the matrix `A` are multiplied by `CTO / CFROM`
+* @param {number} CTO - the matrix `A` are multiplied by `CTO / CFROM`
+* @param {NonNegativeInteger} M - number of rows in matrix `A`
+* @param {NonNegativeInteger} N - number of columns in matrix `A`
+* @param {Float64Array} A - input matrix
+* @param {integer} strideA1 - stride of the first dimension of `A`
+* @param {integer} strideA2 - stride of the second dimension of `A`
+* @param {NonNegativeInteger} offsetA - starting index for `A`
+* @returns {Float64Array} scaled matrix `A`
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] ); // => [ [ 1.0, 2.0 ], [ 3.0, 4.0 ], [ 5.0, 6.0 ] ]
+*
+* dlascl( 'general', 0, 0, 1.0, 2.0, 3, 2, A, 2, 1, 0 );
+* // A => <Float64Array>[ 2.0, 4.0, 6.0, 8.0, 10.0, 12.0 ]
+*/
+function dlascl( type, KL, KU, CFROM, CTO, M, N, A, strideA1, strideA2, offsetA ) { // eslint-disable-line max-params
+	var cfromc;
+	var cfrom1;
+	var ctoc;
+	var cto1;
+	var done;
+	var mul;
+	var da1;
+	var da0;
+	var S1;
+	var S0;
+	var ia;
+	var i0;
+	var i1;
+	var k3;
+	var k4;
+	var k1;
+	var k2;
+	var sh;
+	var sa;
+	var o;
+
+	if ( N === 0 || M === 0 ) {
+		return A;
+	}
+
+	done = false;
+
+	cfromc = CFROM;
+	ctoc = CTO;
+
+	while ( !done ) {
+		cfrom1 = CTO * smlnum;
+		if ( cfrom1 === cfromc ) {
+			// cfromc is Infinity, multiply by a correctly signed zero for finite ctoc or NaN
+			mul = ctoc / cfromc;
+			done = true;
+			cto1 = ctoc;
+		} else {
+			cto1 = ctoc / bignum;
+			if ( cto1 === ctoc ) {
+				// ctoc is either zero or Infinity, thus ctoc itself is a correct multiplication factor
+				mul = ctoc;
+				done = true;
+				cfromc = 1.0;
+			} else if ( abs( cfrom1 ) > abs( ctoc ) && ctoc !== 0.0 ) {
+				mul = smlnum;
+				done = false;
+				ctoc = cto1;
+			} else if ( abs( cto1 ) > abs( cfromc ) ) {
+				mul = bignum;
+				done = false;
+				ctoc = cto1;
+			} else {
+				mul = ctoc / cfromc;
+				done = true;
+			}
+		}
+
+		if ( type === 'general' ) {
+			// Full matrix
+
+			// Resolve the loop interchange order:
+			o = loopOrder( [ M, N ], [ strideA1, strideA2 ] );
+			sh = o.sh;
+			sa = o.sx;
+
+			// Extract loop variables for purposes of loop interchange: dimensions and loop offset (pointer) increments...
+			S0 = sh[ 0 ];
+			S1 = sh[ 1 ];
+			da0 = sa[ 0 ];
+			da1 = sa[ 1 ] - ( S0*sa[0] );
+
+			// Set the pointers to the first indexed elements in the respective matrices...
+			ia = offsetA;
+
+			// Iterate over the matrix dimensions...
+			for ( i1 = 0; i1 < S1; i1++ ) {
+				for ( i0 = 0; i0 < S0; i0++ ) {
+					A[ ia ] *= mul;
+					ia += da0;
+				}
+				ia += da1;
+			}
```

Comment:
```suggestion
			}
			continue;
```

Same comment for below. No need to perform additional branching below if we do not need to.